client/src/VerificationService.ts

```diff
@@ -130,43 +204,48 @@ export class Verifier {
     if (State.updatingGobraTools) return;
     
     // only verify if it is a gobra file or a go file where the verification was manually invoked.
-    if (!fileUri.endsWith(".gobra") && !(fileUri.endsWith(".go") && event == IdeEvents.Manual)) return;
+    const hasNonGobraAndNonGoFiles = fileUris.some(fileUri => !Verifier.isGoOrGobraPath(fileUri.fsPath));
+    const hasGoFiles = fileUris.some(fileUri => fileUri.fsPath.endsWith(".go"));
+    if (hasNonGobraAndNonGoFiles) {
+      return;
+    }
+    if (hasGoFiles && event != IdeEvents.Manual) {
+      return;
+    }
```

Comment:
shouldn't it fail more visibly when `hasNonGobraAndNonGoFiles` is true?

Reply:
Also, what happens if we have a `go.mod` file in the package? Does this set the flag `hasNonGobraAndNonGoFiles`?